redash/handlers/widgets.py

```diff
@@ -52,10 +52,20 @@ def post(self):
 
 class WidgetResource(BaseResource):
     @require_permission('edit_dashboard')
+    def post(self, widget_id):
+        widget = models.Widget.get_by_id_and_org(widget_id, self.current_org)
+        require_admin_or_owner(widget.dashboard.user_id)
+        widget_properties = request.get_json(force=True)
+        widget.text = widget_properties['text']
+        widget.width = widget_properties['width']
```

Comment:
Allowing to change widget size, means that we need to handle the case of the layout needs to change (in case we change it from "Regular" to "Double").

I'm totally fine with limiting the editing functionality to only the text itself (without size).